src/module/SlisBnbModule.sol

```diff
@@ -0,0 +1,138 @@
+// SPDX-License-Identifier: BSD-3-Clause
+pragma solidity ^0.8.24;
+
+import {IStakeManager, IERC20, Math} from "src/Common.sol";
+
+/**
+ * @title SingleModule
+ * @dev Library for Single Asset 4626 Vault module with slisBNB 2024-09-12
+ */
+library SlisBnbModule {
+    using Math for uint256;
+
+    /// @dev The slisBNB StakeManager contract https://github.com/lista-dao/synclub-contracts/tree/v1.1
+    IStakeManager constant stakeManager = IStakeManager(0x1adB950d8bB3dA4bE104211D5AB038628e477fE6);
+
+    /**
+     * @dev Struct to hold ERC4626 storage data from oz/ERC4626Upgradeable.sol
+     * @param _asset The ERC20 asset token.
+     * @param _underlyingDecimals The number of decimals of the underlying asset.
+     */
+    struct ERC4626Storage {
+        IERC20 _asset;
+        uint8 _underlyingDecimals;
+    }
+
+    /**
+     * @dev Retrieves the ERC4626Storage struct from a given storage location.
+     * from oz/ERC4626Upgradeable.sol because its private
+     * @return $ The ERC4626Storage struct stored at the given location.
+     */
+    function _retrieveERC4626Storage() internal pure returns (ERC4626Storage storage $) {
+        assembly {
+            $.slot := 0x0773e532dfede91f04b12a73d3d2acd361424f41f76b4fb79f090161e36b4e00
```

Comment:
nice find here - this is actually the only way to access storage inside a library without passing a storage reference

Reply:
I like how Libraries are pure and consume storage from specified slots like this. I'll come back to this research after deploying the base version. Again, I've reverted back to the basic implementation we had previously and will focus on how to generalize later.

Reply:
Makes sense to keep it minimal for now but good that you explored this direction